### PR TITLE
Explain how to change API identifier for doctrine entities

### PR DESCRIPTION
**Pull request by freiondrej**

### CODE REVIEW

**Comment by GregoireHebert** [comment]
This should target 2.2 right?

File: core/identifiers.md [hunk 1]
```diff
@@ -118,6 +118,44 @@ services:
 
 Your `PersonDataProvider` will now work as expected!
 
+### Changing identifier in a Doctrine entity
```
Comment (alanpoulain) [inline core/identifiers.md:121]:
```suggestion
## Changing Identifier in a Doctrine Entity
```

File: core/identifiers.md [hunk 2]
```diff
@@ -118,6 +118,44 @@ services:
...
+
+If your ApiResource is also a Doctrine entity, you have to unmark its id as identifier:
```
Comment (alanpoulain) [inline core/identifiers.md:123]:
```suggestion
If your resource is also a Doctrine entity and you want to use another identifier as the Doctrine one, you have to unmark it:
```

File: core/identifiers.md [hunk 3]
```diff
@@ -118,6 +118,44 @@ services:
...
+
+```php
+<?php
+namespace App\Entity;
+
+use App\Uuid;
```
Comment (alanpoulain) [inline core/identifiers.md:129]: Missing `ApiResource`.

File: core/identifiers.md [hunk 4]
```diff
@@ -118,6 +118,44 @@ services:
...
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity
+ * @ApiResource
+ */
+final class Person
+{
+
```
Comment (alanpoulain) [inline core/identifiers.md:138]: Line to remove.

File: core/identifiers.md [hunk 5]
```diff
@@ -118,6 +118,44 @@ services:
...
+    /**
+     * @var Uuid
+     * @ApiProperty(identifier=true)
+     */
+    public $code;
+    
+    // ...
+}
+```
+
```
Comment (alanpoulain) [inline core/identifiers.md:158]: Line to remove.

**Comment by alanpoulain** [comment]
@GregoireHebert No it's the right branch.

File: core/identifiers.md (hunk 2 above)
Comment (freiondrej) [reply core/identifiers.md:123]: @alanpoulain I don't think this is the case, is it? I don't want to use it as the Doctrine one, but as the ApiPlatform one.

File: core/identifiers.md (hunk 3 above)
Comment (freiondrej) [reply core/identifiers.md:129]: it is not in the existing example either, should I update both?

File: core/identifiers.md (hunk 4 above)
Comment (freiondrej) [reply core/identifiers.md:138]:
```suggestion

```

File: core/identifiers.md (hunk 3 above)
Comment (alanpoulain) [reply core/identifiers.md:129]: Yes you can :slightly_smiling_face:

File: core/identifiers.md (hunk 2 above)
Comment (alanpoulain) [reply core/identifiers.md:123]: Yes, I made a mistake, updated.

File: core/identifiers.md [hunk 6]
```diff
@@ -118,6 +118,42 @@ services:
...
 
+## Changing Identifier in a Doctrine Entity
+
+If your resource is also a Doctrine entity and you want to use another identifier other than the Doctrine one, you have to unmark it:
+
+```php
+<?php
+namespace App\Entity;
+
+use App\Uuid;
```
Comment (freiondrej) [inline core/identifiers.md:129]:
```suggestion
use ApiPlatform\Core\Annotation\ApiResource;
use App\Uuid;
```

File: core/identifiers.md [hunk 7]
```diff
@@ -118,6 +119,43 @@ services:
...
+     *
+     * @ORM\Id()
+     * @ORM\GeneratedValue()
+     * @ORM\Column(type="integer")
+     */
+    private $id;
+    
+    /**
+     * @var Uuid
+     * @ApiProperty(identifier=true)
```
Comment (GregoireHebert) [inline core/identifiers.md:152]: It should have a column definition too.

Comment (freiondrej) [reply core/identifiers.md:152]: @GregoireHebert thanks, added

**Comment by alanpoulain** [comment]
Thank you @freiondrej!

**Pull request merged**